core/src/main/java/org/apache/iceberg/io/ContentCache.java

```diff
@@ -0,0 +1,300 @@
+            properties,
+            CatalogProperties.IO_CACHE_ENABLED,
+            CatalogProperties.IO_CACHE_ENABLED_DEFAULT);
+    long durationMs =
+        PropertyUtil.propertyAsLong(
+            properties,
+            CatalogProperties.IO_CACHE_EXPIRATION_INTERVAL_MS,
+            CatalogProperties.IO_CACHE_EXPIRATION_INTERVAL_MS_DEFAULT);
+    long totalBytes =
+        PropertyUtil.propertyAsLong(
+            properties,
+            CatalogProperties.IO_CACHE_MAX_TOTAL_BYTES,
+            CatalogProperties.IO_CACHE_MAX_TOTAL_BYTES_DEFAULT);
+    long contentLength =
+        PropertyUtil.propertyAsLong(
+            properties,
+            CatalogProperties.IO_CACHE_MAX_CONTENT_LENGTH,
+            CatalogProperties.IO_CACHE_MAX_CONTENT_LENGTH_DEFAULT);
+
+    if (enabled && durationMs >= 0 && totalBytes > 0 && contentLength > 0) {
+      return new ContentCache(durationMs, totalBytes, contentLength);
+    } else {
+      return null;
+    }
+  }
+
+  private final long expireAfterAccessMs;
+  private final long maxTotalBytes;
+  private final long maxContentLength;
+  private final Cache<String, CacheEntry> cache;
+
+  public ContentCache(long expireAfterAccessMs, long maxTotalBytes, long maxContentLength) {
+    this.expireAfterAccessMs = expireAfterAccessMs;
+    this.maxTotalBytes = maxTotalBytes;
+    this.maxContentLength = maxContentLength;
+
+    Caffeine<Object, Object> builder = Caffeine.newBuilder();
+    if (expireAfterAccessMs > 0) {
+      builder = builder.expireAfterAccess(Duration.ofMillis(expireAfterAccessMs));
+    }
+
+    this.cache =
+        builder
+            .maximumWeight(maxTotalBytes)
+            .weigher(
+                (Weigher<String, CacheEntry>)
+                    (key, value) -> (int) Math.min(value.length, Integer.MAX_VALUE))
+            .recordStats()
+            .build();
+  }
+
+  public long expireAfterAccess() {
+    return expireAfterAccessMs;
+  }
+
+  public long maxContentLength() {
+    return maxContentLength;
+  }
+
+  public long maxTotalBytes() {
+    return maxTotalBytes;
+  }
+
+  public CacheStats stats() {
+    return cache.stats();
+  }
+
+  public CacheEntry get(String key, Function<String, CacheEntry> mappingFunction) {
+    return cache.get(key, mappingFunction);
+  }
+
+  public CacheEntry getIfPresent(String location) {
+    return cache.getIfPresent(location);
+  }
+
+  public InputFile tryCache(FileIO io, String location, long length) {
+    if (length <= maxContentLength) {
+      return new CachingInputFile(this, io, location, length);
+    }
+    return io.newInputFile(location, length);
+  }
+
+  public void invalidate(String key) {
+    cache.invalidate(key);
+  }
+
+  public void invalidateAll() {
+    cache.invalidateAll();
+  }
+
+  public void cleanUp() {
+    cache.cleanUp();
+  }
+
+  public long estimatedCacheSize() {
+    return cache.estimatedSize();
+  }
+
+  @Override
+  public String toString() {
+    return getClass().getSimpleName()
```

Comment:
AFAIK  we tend to use `MoreObjects.toStringHelper(..)` for generating `toString` methods

Reply:
Fixed in 99057de